tracing-subscriber/src/fmt/mod.rs

```diff
@@ -664,6 +742,38 @@ impl Default for Settings {
     }
 }
 
+/// Install a global tracing subscriber that listens for events and
+/// filters based on environment variables, if one is not already set.
+///
+/// Returns whether the initialization was successful.
+///
+/// If the `tracing_log` feature is enabled, this will also install
+/// the LogTracer to convert `Log` records into `tracing` `Event`s.
+pub fn try_init() -> Result<(), impl Error + Send + Sync + 'static> {
+    #[cfg(feature = "tracing_log")]
+    tracing_log::LogTracer::init().map_err(Box::new)?;
+
+    dispatcher::set_global_default(dispatcher::Dispatch::new(
+        Subscriber::builder()
+            .with_env_filter(EnvFilter::from_default_env())
+            .finish(),
```

Comment:
Here's a thought: what do you think about adding `init` and `try_init` methods to the `Builder` type, like this: 
```rust
impl<...> Builder<...> {
    // ...

    pub fn try_init(self) -> Result<(), impl Error + Send + Sync + 'static> {
        #[cfg(feature = "tracing_log")]
        tracing_log::LogTracer::init().map_err(Box::new)?;

        dispatcher::set_global_default(dispatcher::Dispatch::new(self.finish())).map_err(Box::new)
    }

    pub fn init(self) {
        try_init().expect("Unable to install global subscriber")
    }
}
```
The `try_init` free function could then just be 
```rust
pub fn try_init() -> Result<(), impl Error + Send + Sync + 'static> {
    Subscriber::builder()
        .with_env_filter(EnvFilter::from_default_env())
        .try_init()
}
```

Adding `init` and `try_init` to `Builder` as well as having free fns would allow users who _do_ need to make one or two customizations to the subscriber to still write code that's a little less verbose. For example, I could say:
```rust
fmt::Subscriber::builder()
    .with_env_filter(EnvFilter::from_env("MY_CUSTOM_ENV_VAR"))
    .compact()
    .with_writer(my_writer)
    .init();
```
instead of
```rust
let subscriber = fmt::Subscriber::builder()
    .with_env_filter(EnvFilter::from_env("MY_CUSTOM_ENV_VAR"))
    .compact()
    .with_writer(my_writer)
    .finish();
tracing_log::LogTracer::init()?;
dispatcher::set_global_default(dispatcher::Dispatch::new(subscriber))?;
```

Reply:
That sounds like a great idea. 

If I try adding the methods to the Builder [here](https://github.com/tokio-rs/tracing/blob/917e805b73b4b7c956e0ae7c1a55ff9ab85f419b/tracing-subscriber/src/fmt/mod.rs#L400), the compiler complains that N, E, F, and W aren't `Send` + `Sync`. Do you want me to add those trait bounds to each of the parameters?

Reply:
It might be slightly more concise to add a 
```rust
where 
    Subscriber<N, E, F, W>: Send + Sync,
```
bound to that method in particular?

Reply:
Implemented in https://github.com/tokio-rs/tracing/pull/385/commits/817e97b7f2b1a8c7eb14c1de19870e4c79ac5369